pkg/common/plugin/download.go

```diff
@@ -127,9 +127,8 @@ func DownloadPlugin(source PluginSource, targetPath string) error {
 	}
 
 	// download the first blob to the target path
-	blobReference := fmt.Sprintf("%s@%s", source, referenceManifest.Blobs[0].Digest)
-	logrus.Debugf("Downloading blob %s", blobReference)
-	_, blobReader, err := repository.Blobs().FetchReference(ctx, blobReference)
+	logrus.Debugf("Downloading blob %s", referenceManifest.Blobs[0].Digest.String())
```

Comment:
Hi @binbin-li , @akashsinghal , i m not super familiar with this the OCI spec here.  The currently logic access blob[0] directly, I wonder if we need to any null check for referenceManifest.Blobs?

Reply:
good call, I feel null check would be safer.

Reply:
yeah agreed an extra check doesn't hurt. The OCI spec does allow for a 0 layer image so it's possible the blobs could be null or empty. However, by convention we tell users to store their plugin binary as the first layer.

Reply:
thanks all! i ve added a len check